### PR TITLE
Create Role, ServiceAccount and RoleBinding if not exist if serviceAccountName not specified

### PR DESCRIPTION
**Pull request by nikola-jokic**
Closes: #4

### CODE REVIEW

File: controllers/runner_controller.go [hunk 1]
```diff
@@ -78,6 +80,11 @@ type RunnerReconciler struct {
 // +kubebuilder:rbac:groups=actions.summerwind.dev,resources=runners/status,verbs=get;update;patch
 // +kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=core,resources=pods/finalizers,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=pods/exec;rolebindings,verbs=get;create
+// +kubebuilder:rbac:groups=core,resources=pods/log;rolebindings,verbs=get;list;watch
+// +kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=roles;rolebindings,verbs=get;list;watch;create;delete
+// +kubebuilder:rbac:groups=core,resources=serviceaccounts,verbs=get;list;watch;create;delete
+// +kubebuilder:rbac:groups=batch,resources=jobs,verbs=get;list;create;delete
```
Comment (nikola-jokic) [inline controllers/runner_controller.go:87]: In order to create roles for the runner with these permissions, our ClusterRole must have at least the same roles and possibly more. In other words, I cannot assign `batch jobs` role for the runner if the ARC managers ClusterRole does not have the permission as well

File: controllers/runner_controller.go [hunk 2]
```diff
@@ -564,6 +579,140 @@ func mutatePod(pod *corev1.Pod, token string) *corev1.Pod {
...
+	var role rbacv1.Role
+
+	roleNamespacedName := types.NamespacedName{
+		Name:      "job-runner",
+		Namespace: runner.ObjectMeta.Namespace,
+	}
+
+	if err := r.Get(ctx, roleNamespacedName, &role); err != nil {
+		if !kerrors.IsNotFound(err) {
+			// TODO: should we cleanup?
```
Comment (thboop) [inline controllers/runner_controller.go:592]: we are failing on the first create correct? Is there anything to cleanup?

File: controllers/runner_controller.go [hunk 3]
```diff
@@ -564,6 +579,140 @@ func mutatePod(pod *corev1.Pod, token string) *corev1.Pod {
 	return updated
 }
 
+func (r *RunnerReconciler) createJobRunnerRoleIfNotExist(ctx context.Context, runner *v1alpha1.Runner) error {
```
Comment (thboop) [inline controllers/runner_controller.go:582]: Is there going to be a way to clean these up once all runners are deleted from a namespace?

File: config/manager/kustomization.yaml [hunk 4]
```diff
@@ -4,5 +4,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: summerwind/actions-runner-controller
+  newName: jokicnikola07/actions-runner-controller
```
Comment (thboop) [inline config/manager/kustomization.yaml:7]: we will want to revert this before merge!

File: charts/actions-runner-controller/crds/actions.summerwind.dev_runnerdeployments.yaml [hunk 5]
```diff
@@ -4539,7 +4539,7 @@ spec:
...
...                                            description: 'Specifies the object from which to populate the volume with data, if a non-empty volume is desired. This may be any local object from a non-empty API group (non core object) or a PersistentVolumeClaim object. When this field is specified, volume binding will only succeed if the type of the specified object matches some installed volume populator or dynamic provisioner. This field will replace the functionality of the DataSource field and as such if both fields are non-empty, they must have the same value. For backwards compatibility, both fields (DataSource and DataSourceRef) will be set to the same value automatically if one of them is empty and the other is non-empty. There are two important differences between DataSource and DataSourceRef: * While DataSource only allows two specific types of objects, DataSourceRef allows any non-core object, as well as PersistentVolumeClaim objects. * While DataSource ignores disallowed values (dropping them), DataSourceRef preserves all values, and generates an error if a disallowed value is specified. (Alpha) Using this field requires the AnyVolumeDataSource feature gate to be enabled.'
```
Comment (thboop) [inline charts/actions-runner-controller/crds/actions.summerwind.dev_runnerdeployments.yaml:4542]: My generates these different, I'll rerun the generation so the ci behaves.

File: controllers/runner_controller.go (hunk 3 above)
Comment (nikola-jokic) [reply controllers/runner_controller.go:582]: I can list runners in current namespace and remove the role if we just remove the last one

**Pull request closed without merging**